src/shape.rs

```diff
@@ -348,3 +376,356 @@ mod test
         assert_eq!(None, r1.intersect(&r2));
     }
 }
+
+///////////////////////////////////
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+///
+/// Alias for a rectangle with u32 coordinates.
+pub type URoundRect = RoundedRectangle<u32>;
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+///
+/// Alias for a rectangle with i32 coordinates.
+pub type IRoundRect = RoundedRectangle<i32>;
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+///
+/// Alias for a rectangle with f32 coordinates.
+pub type RoundRect = RoundedRectangle<f32>;
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[repr(C)]
+pub struct RoundedRectangle<T = f32>
+{
+    top_left: Vector2<T>,
+    bottom_right: Vector2<T>,
+    radius: T
```

Comment:
Mismatch with the comments above -- since this is type `T`, it might be an int:

> Two points and a float are stored

I'm not sure whether it's best to use `T` here, or an `f32`. I suppose a third option would be to add another (optional) generic parameter `<R = f32>` for the radius, maybe that's best for maximum flexibility in future.

Reply:
Should be fixed, I just changed it from "float" to "a T value" and kept the generic T.